src/charm_state.py

```diff
@@ -266,6 +268,43 @@ def userids_to_list(cls, value: str) -> typing.List[str]:
                 raise ValidationError(f"Invalid user ID format: {user_id}", cls)
         return value_list
 
+    @validator("ready_check")
+    @classmethod
+    def to_pebble_check(cls, value: str) -> typing.Dict[str, typing.Union[str, int]]:
+        """Convert the ready_check field to pebble check.
+
+        Args:
+            value: the input value.
+
+        Returns:
+            The pebble check.
+
+        Raises:
+            ValidationError: if ready_check is invalid.
+        """
+        # expected
+        # period,threshold,timeout
+        config_values = value.split(",")
+        if len(config_values) != 3:
+            raise ValidationError(f"Invalid ready_check, less or more than 3 values: {value}", cls)
```

Comment:
🤔 a trade-off to consider:

a simpler way to write this would be

`foo, bar, baz = value.split(",")`

Which raises a ValueError on wrong number of commas, and pydantic.ValidationError has ValueError as a base class.

Ofc, that would not provide a neat, user-readable exception...

Reply:
Thanks, I'll leave it in favor of the user-readable exception. This configuration is a workaround until https://github.com/canonical/pebble/issues/145 is in place